src/gui_common/AddWindowReorderingSupportToSiblings.cs

```diff
@@ -394,6 +403,55 @@ private void OnWindowReorder(CustomDialog window)
         window.SetAsToplevel(isSetAsToplevel);
     }
 
+    private void OnWindowOpen(CustomDialog window)
+    {
+        if (justOpenedWindows.Contains(window))
+        {
+            // This window is already queued to be opened
+            return;
+        }
+
+        if (!reorderOpenedWindowsQueued)
+        {
+            // Tell the system that there is an opened window and wait in case more windows will be opened at once
+            reorderOpenedWindowsQueued = true;
+            Invoke.Instance.QueueForObject(ReorderOpenedWindows, this);
+        }
+
+        justOpenedWindows.Add(window);
+    }
+
+    private void ReorderOpenedWindows()
+    {
+        try
+        {
+            // Sort the windows to make sure they are updated in the right order
+            justOpenedWindows.Sort((first, second) =>
+            {
+                return connectedWindows[first].GetIndex().CompareTo(connectedWindows[second].GetIndex());
+            });
+        }
+        catch (InvalidOperationException)
```

Comment:
Might be better to just catch all, that way it's more robust.

Reply:
This made me think that we should also check if the connectedWindows dictionary contains the windows when an exception is thrown so I will add that too.